wrangler-service/src/main/java/io/cdap/wrangler/service/common/AbstractWranglerHandler.java

```diff
@@ -205,12 +207,22 @@ protected void respond(HttpServiceResponder responder, String namespace,
     } catch (JsonSyntaxException e) {
       responder.sendJson(HttpURLConnection.HTTP_BAD_REQUEST,
                          new io.cdap.wrangler.proto.workspace.v2.ServiceResponse<>(e.getMessage()));
+    } catch (RemoteExecutionException e) {
+      responder.sendJson(getErrorCode(e.getCause().getRemoteExceptionClassName()),
+                         new io.cdap.wrangler.proto.workspace.v2.ServiceResponse<>(e.getMessage()));
     } catch (Throwable t) {
       responder.sendJson(HttpURLConnection.HTTP_INTERNAL_ERROR,
                          new io.cdap.wrangler.proto.workspace.v2.ServiceResponse<>((t.getMessage())));
     }
   }
 
+  private int getErrorCode(String exceptionName) {
+    boolean isBadRequest = Arrays
+      .asList(BadRequestException.class, ErrorRecordsException.class, JsonSyntaxException.class).
+        stream().anyMatch(e -> e.getName().equals(exceptionName));
+    return isBadRequest ? HttpURLConnection.HTTP_BAD_REQUEST : HttpURLConnection.HTTP_INTERNAL_ERROR;
```

Comment:
I don't think this is the only bad code returned, I believe 404 is returned if a workspace is not found, it is better to extend this to support more error codes

Reply:
The workspace code is still executed locally, so it will not be part of the RemoteExecutionException. Will test to verify this.

Reply:
Tested to verify that 404 is returned when workspace is not found.